src/kv/kv.h

```diff
@@ -1627,15 +1683,22 @@ namespace kv
           if (success_ != CommitSuccess::OK)
             LOG_DEBUG_FMT("Failed Tx commit {}", last_replicated + offset);
 
+          std::vector<uint8_t> all_data;
```

Comment:
concatenating `replicated` and `derived` write sets here instead of inside `history` as we are passing all of the serialised data to `replicate()`